### PR TITLE
Module 5

### PR DESCRIPTION
**Pull request by JoeFisherW**
Getting a json decoder error but the app works locally.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM python:3.8.2-buster as base
+
+RUN pip install gunicorn
```
Comment (Jonesey13) [inline Dockerfile:3]: Gunicorn can be managed by poetry

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+RUN pip install poetry
```
Comment (Jonesey13) [inline Dockerfile:4]: Poetry should be installed via the recommended method:
image

File: pyproject.toml [hunk 3]
```diff
@@ -12,6 +12,7 @@ python-dateutil = "^2.8.1"
 flask = "^1.1.2"
 requests = "^2.24.0"
 python-dotenv = "^0.12.0"
+gunicorn = "20.0.4"
```
Comment (Jonesey13) [inline pyproject.toml:15]: Ah you do have it!

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
This still needs a few changes to pass review:
- The README.md needs a re-write to use poetry, explain how to setup the flask env variables and how to build and run the docker containers
- You don't have a `.dockerignore` file so your trello credentials are being baked into your images

File: .dockerignore [hunk 4]
```diff
@@ -0,0 +1 @@
+.env
```
Comment (hugh-emerson) [inline .dockerignore:1]: This is what was *needed*, but you could also optimise the images by ignoring some more things. Like everything in .gitignore, `.gitignore` itself, `.git`

File: Dockerfile (hunk 1 above)
Comment (hugh-emerson) [reply Dockerfile:3]: As Alex pointed out, you do have gunicorn listed in your pyproject.toml file. So you can remove this line.

File: README.md [hunk 5]
```diff
@@ -2,23 +2,38 @@
...
 ```
-### On Windows (Using Git Bash)
-```bash
-$ source setup.sh --windows
+    * Key
+    * TOKEN
+    * BoardID
+    * ToDoId (List Id)
+    * DoingId (List Id)
+    * DoneId (List Id)
```
Comment (hugh-emerson) [inline README.md:19]: For slightly more convenience, these could be included in .env.template instead.

File: README.md [hunk 6]
```diff
@@ -2,23 +2,38 @@
 
 ## Getting started
 
-The project uses a virtual environment to isolate package dependencies. To create the virtual environment and install required packages, run the following from a bash shell terminal:
+The project uses poetry to manage package dependencies.
+This can be setup using:
+``` bash
+     $ run poetry install
```
Comment (hugh-emerson) [inline README.md]:
```suggestion
     $ poetry install
```

File: README.md [hunk 7]
```diff
@@ -2,23 +2,38 @@
 
 ## Getting started
 
-The project uses a virtual environment to isolate package dependencies. To create the virtual environment and install required packages, run the following from a bash shell terminal:
+The project uses poetry to manage package dependencies.
+This can be setup using:
```
Comment (hugh-emerson) [inline README.md:6]: If the reader has already installed Poetry.

File: README.md [hunk 8]
```diff
@@ -2,23 +2,38 @@
...
+    * TOKEN
+    * BoardID
+    * ToDoId (List Id)
+    * DoingId (List Id)
+    * DoneId (List Id)
 ```
 
-Once the setup script has completed and all packages have been installed, start the Flask app by running:
-```bash
-$ flask run
```
Comment (hugh-emerson) [inline README.md:18]: Your README now implies the reader *has to* use Docker. But they can still run the app locally with `poetry run flask run`.

**Review by hugh-emerson — APPROVED** [review]
I've left some suggestions, mainly clarifying the README. But you can merge this whenever you're ready.

**Comment by hugh-emerson** [comment]
Ah, also, requirements.txt, requirements-frozen.txt and setup.sh should all be deleted as you are now using poetry instead of pip.

**Pull request merged**